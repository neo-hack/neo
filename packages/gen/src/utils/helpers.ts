import { Workflow } from '../interface'
import { hooks } from '../utils/hooks'

import Observable from 'zen-observable'
import type Listr from 'listr'
import { LIFE_CYCLES } from 'src/constants'

/**
 * @description convert schema to listr
 */
export const toListr = (schema: Workflow) => {
  const tasks: Listr.ListrTask[] = Object.keys(schema.jobs || {}).map((key) => {
    const taskName = schema.jobs![key].name || key
    return {
      title: taskName,
      task: () => {
        let ob: ZenObservable.SubscriptionObserver<any>
        const OB = new Observable<any>((observer) => {
          ob = observer
        })
        hooks.addHooks({
          [LIFE_CYCLES.STEP]: (args: { step: string }) => {
            ob.next(args.step)
          },
          [taskName]: () => {
            ob.complete()
          },
        })
        return OB as any
      },
    }
  })
  return tasks
}
