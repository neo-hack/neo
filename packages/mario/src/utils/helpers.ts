import { Workflow } from '../interface'
import { hooks } from './hooks'

import Observable from 'zen-observable'
import type Listr from 'listr'
import { LIFE_CYCLES } from 'src/constants'

/**
 * @description convert schema to listr
 */
export const toListr = (schema: Workflow) => {
  const tasks: Listr.ListrTask[] = Object.keys(schema.jobs || {}).map((key) => {
    const taskName = schema.jobs![key].name || key
    return {
      title: taskName,
      task: () => {
        let ob: ZenObservable.SubscriptionObserver<any>
        const OB = new Observable<any>((observer) => {
          ob = observer
        })
        hooks.addHooks({
          [LIFE_CYCLES.STEP]: (args: { step: string }) => {
            ob.next(args.step)
          },
          [taskName]: (args: { error: Error }) => {
            if (!args.error) {
              ob.complete()
              return
            }
            ob.error(args.error)
          },
        })
        return OB as any
      },
    }
  })
  return tasks
}
