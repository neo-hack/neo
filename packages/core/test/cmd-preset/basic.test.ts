// import nock from 'nock'
import tempy from 'tempy'
import path from 'path'
import debug from 'debug'

import { execNeo } from '../helpers'

// const registry = `http://localhost:${REGISTRY_MOCK_PORT}`
const storeDir = path.join(tempy.directory(), '.store')
debug.enable('neo:lockfile*')

it('load preset npm package should work', async () => {
  const { stdout, stderr } = await execNeo(['preset', '@aiou/preset-aiou', '--store-dir', storeDir])
  // nock(registry)
  //   .get('/@aiou/ts-lib-template/-/ts-lib-template-0.5.0.tgz')
  //   .replyWithFile(200, r('test/assets/ts-lib-template-0.5.0.tgz'))
  expect(stdout).toMatchSnapshot()
  expect(stderr).toBe('')
})
