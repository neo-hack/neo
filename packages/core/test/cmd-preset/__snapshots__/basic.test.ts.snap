// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load preset npm package should work 1`] = `
"{
  id: 'registry.npmjs.org/@aiou/preset-aiou/0.0.1',
  isLocal: false,
  isInstallable: true,
  latest: '0.0.1',
  manifest: {
    name: '@aiou/preset-aiou',
    version: '0.0.1',
    dist: {
      integrity: 'sha512-+X0RUSXy9FzkC61U28uqglPg9h152yadRmST/71ykbkN13gxi2gVN1M4uTo+bb5os/7CDO7EnLBprYZxwQR7Sg==',
      shasum: '1b226a671df8c1bdc9b2f35945fb4282af356dd0',
      tarball: 'https://registry.npmjs.org/@aiou/preset-aiou/-/preset-aiou-0.0.1.tgz',
      fileCount: 5,
      unpackedSize: 2086,
      'npm-signature': '-----BEGIN PGP SIGNATURE-----\\\\r\\\\n' +
        'Version: OpenPGP.js v3.0.13\\\\r\\\\n' +
        'Comment: https://openpgpjs.org\\\\r\\\\n' +
        '\\\\r\\\\n' +
        'wsFcBAEBCAAQBQJhsr+VCRA9TVsSAnZWagAA+a4P/joZub2nzPPcGT/+4kxO\\\\n' +
        'Mk6t7c20m+JFwE+mwIr6XXocuZIdeO4O5PxOWVTreeezzpZwuPuR6+4vVwGp\\\\n' +
        'GpEk8qWcWNlGGfk761gqk2M+Mf/MVinP5nZ8dmWbcMR8hSHNpH4dULzUi1FA\\\\n' +
        'dEMQBMR+5Omj4/xfUAVLLN6YPCB3QPaQxvBhWGQHMP7LndAjD5qgNeh+gJ03\\\\n' +
        'F3R6P3genLIsRh+R6G3tApTvMAr0pcOGlWHVQcvHTelOc9uJf1ebNEYZ38/J\\\\n' +
        '1eNFfeAb+RBLSM37h+kShVDr+JtUfbHjTCud/OKYD5BRmHdNMr7G/DCMuXBU\\\\n' +
        'gQPjmPfKtFn/TjabRuK8VE+r4jKlnwKdGNJRM/vRtAN/l38LAF4WtB19X9mE\\\\n' +
        'R4eY7rfYJPfSTTytdIUTonXIJu4Ayt2ZSPz+ryOWDJXymU6nj8oP/bDDUMXj\\\\n' +
        'G8EGF4qg7tWGu+KLDRH4St/OH0DFgzjGQqwZtUwjPVYG+CqoYXHJadwKu8L1\\\\n' +
        'tA8a//OGZlSr8gZXpdOB+wy1JjpCQxepHlvhD3/hTXm1C5WFP8OWhAN6006N\\\\n' +
        'zj99PRLJV/WmDnj7AMtcyQGW1icWhUjtdwsum2UgHEMNwf4YXSCr9Sqf/chp\\\\n' +
        '1MfWtWBxtFRzFIPbyW8ZSmY8IOK+icllRAFpwH5rbVSCFBqMQL4/1S8mq7av\\\\n' +
        'x16f\\\\r\\\\n' +
        '=GFCJ\\\\r\\\\n' +
        '-----END PGP SIGNATURE-----\\\\r\\\\n'
    }
  },
  normalizedPref: undefined,
  resolution: {
    integrity: 'sha512-+X0RUSXy9FzkC61U28uqglPg9h152yadRmST/71ykbkN13gxi2gVN1M4uTo+bb5os/7CDO7EnLBprYZxwQR7Sg==',
    registry: 'https://registry.npmjs.org/',
    tarball: 'https://registry.npmjs.org/@aiou/preset-aiou/-/preset-aiou-0.0.1.tgz'
  },
  resolvedVia: 'npm-registry',
  updated: true
}"
`;
